src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
+            {
+                try
+                {
+                    object result = CachedExpressionCompiler.Process(expression)((TModel)container);
+                    if (expression.ReturnType.GetTypeInfo().IsEnum)
+                    {
+                        var enumValue = metadata.EnumNamesAndValues[result.ToString()];
+                        foreach (var kvp in metadata.EnumGroupedDisplayNamesAndValues)
+                        {
+                            if (kvp.Value == enumValue)
```

Comment:
Always use `string.Equals(...)` when doing `string` comparisons. Also be explicit about the `StringComparison`.